pkg/kudoctl/packages/package.go

```diff
@@ -1,186 +1,20 @@
-	isOperatorFile := func(name string) bool {
-		return strings.HasSuffix(name, operatorFileName)
-	}
-
-	isTemplateFile := func(name string) bool {
-		matched, err := regexp.Match(templateFileNameRegex, []byte(name))
-		if err != nil {
-			panic(err)
-		}
-		return matched
-	}
-
-	isParametersFile := func(name string) bool {
-		return strings.HasSuffix(name, paramsFileName)
-	}
-
-	switch {
-	case isOperatorFile(filePath):
-		if err := yaml.Unmarshal(fileBytes, &currentPackage.Operator); err != nil {
-			return errors.Wrap(err, "failed to unmarshal operator file")
-		}
-		if currentPackage.Operator.APIVersion == "" {
-			currentPackage.Operator.APIVersion = APIVersion
-		}
-		if currentPackage.Operator.APIVersion != APIVersion {
-			return fmt.Errorf("expecting supported API version %s but got %s", APIVersion, currentPackage.Operator.APIVersion)
-		}
-	case isTemplateFile(filePath):
-		pathParts := strings.Split(filePath, "templates/")
-		name := pathParts[len(pathParts)-1]
-		currentPackage.Templates[name] = string(fileBytes)
-	case isParametersFile(filePath):
-		paramsFile, err := readParametersFile(fileBytes)
-		if err != nil {
-			return errors.Wrapf(err, "failed to unmarshal parameters file: %s", filePath)
-		}
-		currentPackage.Params = make([]v1beta1.Parameter, 0)
-		defaultRequired := true
-		for _, param := range paramsFile.Params {
-			if param.Required == nil {
-				// applying default value of required for all params where not specified
-				param.Required = &defaultRequired
-			}
-			currentPackage.Params = append(currentPackage.Params, param)
-		}
-	default:
-		return fmt.Errorf("unexpected file when reading package from filesystem: %s", filePath)
-	}
-	return nil
-}
-
-func readParametersFile(fileBytes []byte) (ParametersFile, error) {
-	paramsFile := ParametersFile{}
-	if err := yaml.Unmarshal(fileBytes, &paramsFile); err != nil {
-		return paramsFile, err
-	}
-	if paramsFile.APIVersion == "" {
-		paramsFile.APIVersion = APIVersion
-	}
-	if paramsFile.APIVersion != APIVersion {
-		return paramsFile, fmt.Errorf("expecting supported API version %s but got %s", APIVersion, paramsFile.APIVersion)
-	}
-
-	return paramsFile, nil
-}
-
-func newPackageFiles() PackageFiles {
-	return PackageFiles{
-		Templates: make(map[string]string),
-	}
-}
-
-func validateTask(t v1beta1.Task, templates map[string]string) []string {
-	var resources []string
-	switch t.Kind {
-	case task.ApplyTaskKind:
-		resources = t.Spec.ResourceTaskSpec.Resources
-	case task.DeleteTaskKind:
-		resources = t.Spec.ResourceTaskSpec.Resources
-	case task.DummyTaskKind:
-	default:
-		log.Printf("no validation for task kind %s implemented", t.Kind)
-	}
-
-	var errs []string
-	for _, res := range resources {
-		if _, ok := templates[res]; !ok {
-			errs = append(errs, fmt.Sprintf("task %s missing template: %s", t.Name, res))
-		}
-	}
-
-	return errs
-}
-
-func (p *PackageFiles) getCRDs() (*Resources, error) {
+func (p *Files) Resources() (*Resources, error) {
```

Comment:
I am kind of struggling with this file but I don't have a better answer right now :)

Reply:
That makes two of us. I wanted to put this file somewhere else but failed to find the place 🤷‍♂

Reply:
actually, moving this file is also not without consequences as `func (p *Files) Resources()` can only exist in the same package where `Files` is defined. So we'll have to change the method signature to smth. like `func Resources(p *Files)`... I'm inclined to leave as is